src/io/flutter/view/preview/PreviewView.java

```diff
@@ -0,0 +1,343 @@
+    final DartAnalysisServerServiceEx analysisServiceEx = DartAnalysisServerServiceEx.get(analysisService);
+    flutterAnalysisServer = new FlutterDartAnalysisServer(analysisServiceEx);
+
+    final MessageBusConnection bus = project.getMessageBus().connect(project);
+    bus.subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, new FileEditorManagerListener() {
+      @Override
+      public void selectionChanged(@NotNull FileEditorManagerEvent event) {
+        if (currentFile != null) {
+          flutterAnalysisServer.removeOutlineListener(currentFile.getPath(), outlineListener);
+          currentFile = null;
+          currentEditor = null;
+        }
+        if (event.getNewFile() != null && event.getNewEditor() instanceof TextEditor) {
+          currentFile = event.getNewFile();
+          currentEditor = (TextEditor)event.getNewEditor();
+          flutterAnalysisServer.addOutlineListener(currentFile.getPath(), outlineListener);
+        }
+      }
+    });
+  }
+
+  @Override
+  public void dispose() {
+  }
+
+  @NotNull
+  @Override
+  public PreviewView.State getState() {
+    return this.state;
+  }
+
+  @Override
+  public void loadState(PreviewView.State state) {
+    this.state = state;
+  }
+
+  public void initToolWindow(@NotNull ToolWindow toolWindow) {
+    final ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
+    final ContentManager contentManager = toolWindow.getContentManager();
+
+    final DefaultActionGroup toolbarGroup = new DefaultActionGroup();
+
+    toolbarGroup.add(new AnAction(FlutterIcons.PreviewSurroundCenter) {
+      @Override
+      public void actionPerformed(AnActionEvent e) {
+        Messages.showErrorDialog("Not implemented yet.", "TODO");
+      }
+    });
+
+    toolbarGroup.add(new AnAction(FlutterIcons.PreviewSurroundColumn) {
+      @Override
+      public void actionPerformed(AnActionEvent e) {
+        Messages.showErrorDialog("Not implemented yet.", "TODO");
+      }
+    });
+
+    toolbarGroup.add(new AnAction(FlutterIcons.PreviewSurroundRow) {
+      @Override
+      public void actionPerformed(AnActionEvent e) {
+        Messages.showErrorDialog("Not implemented yet.", "TODO");
+      }
+    });
+
+
+    final Content content = contentFactory.createContent(null, null, false);
+    content.setCloseable(false);
+
+    final SimpleToolWindowPanel windowPanel = new SimpleToolWindowPanel(true, true);
+    content.setComponent(windowPanel);
+
+    windowPanel.setToolbar(ActionManager.getInstance().createActionToolbar("PreviewViewToolbar", toolbarGroup, true).getComponent());
+
+    final DefaultMutableTreeNode rootNode = new DefaultMutableTreeNode();
+
+    tree = new OutlineTree(rootNode);
+    tree.setCellRenderer(new OutlineTreeCellRenderer());
+    tree.expandAll();
+
+    tree.addMouseListener(new MouseAdapter() {
+      @Override
+      public void mouseClicked(MouseEvent e) {
+        if (e.getClickCount() == 2) {
```

Comment:
Is there no double click accessor?

Reply:
No, I think this is how it is done in Swing.
On the bright side, we can also have triple clicks :-)